frontend/src/concepts/pipelines/topology/parseUtils.ts

```diff
@@ -159,26 +161,133 @@ export const parseRuntimeInfo = (
     startTime: thisTaskDetail[0].start_time,
     completeTime: thisTaskDetail[0].end_time,
     state: lowestProgress(thisTaskDetail),
-    taskId: thisTaskDetail[0].task_id,
+    taskId: `task.${taskId}`,
     podName: thisTaskDetail[0].child_tasks?.find((o) => o.pod_name)?.pod_name,
   };
 };
 
-export const translateStatusForNode = (stateKF?: RuntimeStateKF): RunStatus | undefined => {
-  switch (stateKF) {
+export const parseRuntimeInfoFromExecutions = (
+  taskId: string,
+  executions?: Execution[] | null,
+): PipelineTaskRunStatus | undefined => {
+  if (!executions) {
+    return undefined;
+  }
+
+  const execution = executions.find(
+    (e) => e.getCustomPropertiesMap().get('task_name')?.getStringValue() === taskId,
+  );
+
+  if (!execution) {
+    return undefined;
+  }
+
+  const lastUpdatedTime = execution.getLastUpdateTimeSinceEpoch();
+  let completeTime;
+  const lastKnownState = execution.getLastKnownState();
+  // Logic comes from https://github.com/opendatahub-io/data-science-pipelines/blob/master/frontend/src/components/tabs/RuntimeNodeDetailsV2.tsx#L245-L253
+  if (
+    lastUpdatedTime &&
+    (lastKnownState === Execution.State.COMPLETE ||
+      lastKnownState === Execution.State.FAILED ||
+      lastKnownState === Execution.State.CACHED ||
+      lastKnownState === Execution.State.CANCELED)
+  ) {
+    completeTime = new Date(lastUpdatedTime).toISOString();
+  }
+  return {
+    startTime: new Date(execution.getCreateTimeSinceEpoch()).toISOString(),
+    completeTime,
+    state: getResourceStateText({
+      resourceType: ResourceType.EXECUTION,
+      resource: execution,
+    }),
+    taskId: `task.${taskId}`,
+    podName: execution.getCustomPropertiesMap().get('pod_name')?.getStringValue(),
+  };
+};
+
+export enum ResourceType {
+  ARTIFACT = 'ARTIFACT',
+  EXECUTION = 'EXECUTION',
+}
+
+export interface ArtifactProps {
+  resourceType: ResourceType.ARTIFACT;
+  resource: Artifact;
+}
+
+export interface ExecutionProps {
+  resourceType: ResourceType.EXECUTION;
+  resource: Execution;
+}
+
+export type ResourceInfoProps = ArtifactProps | ExecutionProps;
+
+// Get text representation of resource state.
+// Works for both artifact and execution.
+export const getResourceStateText = (
+  props: ResourceInfoProps,
+): ArtifactStateKF | ExecutionStateKF | undefined => {
+  if (props.resourceType === ResourceType.ARTIFACT) {
+    const state = props.resource.getState();
```

Comment:
This `Artifact` switch doesn't appear to be unit tested.

Reply:
nit: The default case is not tested, but the bulk was done here so not a blocker.